### PR TITLE
[WIKI] Study Parameters - Subprojects

### PR DESCRIPTION
**Pull request by ridz1208**
### Brief summary of changes

How to add/configure Subprojects

new PR to replace #4307 
___

*This and all upcoming PRs for the WIKI will contain one or a couple files at most. this will hopefully prevent PRs from staying open for very long periods.*

*This and all upcoming PRs are open to discussions and suggestions. Most of the content has been copied over from the existing WIKI with sometimes minor, major or total changes.*

*Feel free to make suggestions and/or push directly to the branch !!

### CODE REVIEW

File: docs/wiki/01 - STUDY PARAMETERS SETUP/01 - Study Variables/04 - Subprojects.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+### Projects
+**Subprojects** must be associated to at least one Project in order to be able to 
+create timepoints for candidates. This association should be defined directly on the 
+front end through the Configuration module.
+
+> Note: the only way to view a list of all the projects affiliated to a subproject 
+is via the MySQL back-end. 
+
+Sometimes it's useful to add sub  project-project affiliations directly in the MySQL
```
Comment (driusan) [inline docs/wiki/01 - STUDY PARAMETERS SETUP/01 - Study Variables/04 - Subprojects.md]:
```suggestion
Sometimes it's useful to add project-subproject affiliations directly in the MySQL 
```

typo, I think?

Comment (ridz1208) [reply docs/wiki/01 - STUDY PARAMETERS SETUP/01 - Study Variables/04 - Subprojects.md]: you think correctly

File: docs/wiki/99 - Developers/SQL Dictionary.md [hunk 2]
```diff
@@ -47,3 +47,37 @@
 	| `OptimumMaxDays `    | Candidate's maximum age in days for visit to be flagged as _within **optimal** parameters of the study_   | The only effect of this field is a YES/NO flag showing up on the instrument_list page|
 	| `WindowMidpointDays `| Candidate's ideal age in days for a visit                                                                 |  |
 	
+
```
Comment (maltheism) [inline docs/wiki/99 - Developers/SQL Dictionary.md]: @ridz1208 The table is nicely done. Did you use anything to generate it or did you type it all manually?

File: docs/wiki/99 - Developers/SQL Dictionary.md [hunk 3]
```diff
@@ -47,3 +47,37 @@
...
+- Table: `subproject`
+ 
+ 	*This table stores the list of subprojects configured for the study.*
+ 
+	|        Field       |                  Description                  |                                     Notes                                    |
+	|:------------------:|:---------------------------------------------:|:----------------------------------------------------------------------------:|
+	| `SubprojectID`     | Identifier of the subproject                  | Avoid setting this field explicitly when inserting data, it auto increments. |
+	| `title`            | Name of the subproject                        |                                                                              |
+	| `useEDC`           | Use the Expected date Of Confinement          |                                                                              |
+	| `WindowDifference` |                                               | Deprecated ??                                                                |
```
Comment (driusan) [inline docs/wiki/99 - Developers/SQL Dictionary.md]: This was used to determine if the error flagging for the candidate age on an instrument should be based on the 'optimal' window in the visit_windows table, or if it should be done by guessing what the nearest test battery is and assuming that it should have been in that window. As far as I know the code hasn't been removed or deprecated.

File: docs/wiki/99 - Developers/SQL Dictionary.md [hunk 4]
```diff
@@ -47,3 +47,37 @@
...
+	| `RecruitmentTarget`| Expected number of candidates to be recruited |                                                                              |
+	
+	
+- Table: `project_subproject_rel`
+ 
+ 	*This table stores the association of projects with subprojects*
+ 
+	|            Field          |                    Description                   |                                     Notes                                    |
+	|:-------------------------:|:------------------------------------------------:|:----------------------------------------------------------------------------:|
+	| `ProjectSubprojectRelID ` | Identifier of the project-subproject relation    | Avoid setting this field explicitly when inserting data, it auto increments. |
```
Comment (driusan) [inline docs/wiki/99 - Developers/SQL Dictionary.md]: Why does this field exist for a rel table? It should probably be explained in the notes.

File: docs/wiki/99 - Developers/SQL Dictionary.md (hunk 2 above)
Comment (ridz1208) [reply docs/wiki/99 - Developers/SQL Dictionary.md]: manually

File: docs/wiki/99 - Developers/SQL Dictionary.md (hunk 3 above)
Comment (ridz1208) [reply docs/wiki/99 - Developers/SQL Dictionary.md]: i doont see it being used anywhere. can you point me to the code ?

File: docs/wiki/99 - Developers/SQL Dictionary.md (hunk 4 above)
Comment (ridz1208) [reply docs/wiki/99 - Developers/SQL Dictionary.md]: it's to be able to point to a single row from another rel table
the visit_project_subproject_rel

Comment (driusan) [reply docs/wiki/99 - Developers/SQL Dictionary.md]: just a note that other tables need it as a foreign key would be enough, I think.

File: docs/wiki/99 - Developers/SQL Dictionary.md (hunk 3 above)
Comment (driusan) [reply docs/wiki/99 - Developers/SQL Dictionary.md]: I can't find it, I guess you can leave it marked as deprecated. Can you remove the `??`?

File: docs/wiki/99 - Developers/SQL Dictionary.md [hunk 5]
```diff
@@ -47,3 +47,37 @@
...
+	| `RecruitmentTarget`| Expected number of candidates to be recruited |                                                                              |
+	
+	
+- Table: `project_subproject_rel`
+ 
+ 	*This table stores the association of projects with subprojects*
+ 
+	|            Field          |                    Description                   |                                     Notes                                    |
+	|:-------------------------:|:------------------------------------------------:|:----------------------------------------------------------------------------:|
+	| `ProjectSubprojectRelID ` | Identifier of the project-subproject relation    | Avoid setting this field explicitly when inserting data, it auto increments. Other tables in the database require this field as a foreign key reference to point o a specific project-subproject tuple.   |
```
Comment (driusan) [inline docs/wiki/99 - Developers/SQL Dictionary.md]:
```suggestion
	| `ProjectSubprojectRelID ` | Identifier of the project-subproject relation    | Avoid setting this field explicitly when inserting data, it auto increments. Other tables in the database require this field as a foreign key reference to point to a specific project-subproject tuple.   |
```

File: docs/wiki/99 - Developers/SQL Dictionary.md [hunk 6]
```diff
@@ -47,3 +47,37 @@
...
+- Table: `subproject`
+ 
+ 	*This table stores the list of subprojects configured for the study.*
+ 
+	|        Field       |                  Description                  |                                     Notes                                    |
+	|:------------------:|:---------------------------------------------:|:----------------------------------------------------------------------------:|
+	| `SubprojectID`     | Identifier of the subproject                  | Avoid setting this field explicitly when inserting data, it auto increments. |
+	| `title`            | Name of the subproject                        |                                                                              |
+	| `useEDC`           | Use the Expected date Of Confinement          |                                                                              |
+	| `WindowDifference` |                                               | Deprecated ??                                                                |
```
Comment (driusan) [inline docs/wiki/99 - Developers/SQL Dictionary.md]:
```suggestion
	| `WindowDifference` |                                               | Deprecated                                                                |
```

**Pull request merged**